packages/babel-parser/src/plugins/flow/index.js

```diff
@@ -2686,10 +2686,14 @@ export default (superClass: Class<Parser>): Class<Parser> =>
           );
         }
         /*:: invariant(firstIdent instanceof N.Node) */
-        isBinding = true;
         specifier.imported = firstIdent;
         specifier.importKind = null;
-        specifier.local = cloneIdentifier(specifier.imported);
+        if (this.eatContextual("as")) {
+          specifier.local = this.parseIdentifier();
+        } else {
+          isBinding = true;
+          specifier.local = specifier.imported.__clone();
```

Comment:
We are moving away from `__clone()` (https://github.com/babel/babel/commit/d3a7cd5e8d5f99fb0b465a5c1d7209d5e1e6ab03). Can you use this instead?
```suggestion
          specifier.local = cloneIdentifier(specifier.imported);
```

Also, do you think we can merge these lines with lines 2673-2678? Maybe by doing something like this:
```js
if (this.isContextual("as") && !this.isLookaheadContextual("as")) { // Line 2649
  // ...
} else {
  if (specifierTypeKind !== null && (this.match(tt.name) || this.state.type.keyword)) {
    // `import {type foo`
    specifier.imported = this.parseIdentifier(true);
    specifier.importKind = specifierTypeKind;
  } else {
    specifier.imported = firstIdent;
    specifier.importKind = null;
  }
  
  if (this.eatContextual("as")) {
    // ...
  } else {
    // ...
  }
}
```

Reply:
Merged the lines as you suggested. Thank you.